src/coreComponents/common/format/table/TableLayout.hpp

```diff
@@ -51,68 +55,411 @@ class TableLayout
+     * @param name The name to set for the column.
+     * @return The current column object.
+     */
+    Column & setName( string_view name );
+
+    /**
+     * @brief Set the column visibility.
+     * @param celltype Cell type to apply to hide the colmun
+     * @return The current column .
+     */
+    Column & setVisibility( CellType celltype );
+
+    /**
+     * @brief Adds multiple sub-columns to the column.
+     * @param subCol A list of sub-column names to add.
+     * @return The current column object
+     */
+    TableLayout::Column & addSubColumns( std::initializer_list< TableLayout::Column > subCol );
+
+    /**
+     * @brief Adds multiple sub-columns to the column.
+     * @param subColName A list of sub-column names to add.
+     * @return The current column object
+     */
+    TableLayout::Column & addSubColumns( std::initializer_list< string > subColName );
+
+    /**
+     * @brief Adds a single sub-column to the column.
+     * @param subColName The name of the sub-column to add.
+     * @return The current column object.
+     */
+    TableLayout::Column & addSubColumns( string const & subColName );
+
+    /**
+     * @brief Sets the header alignment for the column.
+     * @param headerAlignment The alignment to set for the column header (left, right, or center).
+     * @return The current column object
+     */
+    TableLayout::Column & setHeaderAlignment( Alignment headerAlignment );
+
+    /**
+     * @brief Sets the values alignment for the column.
+     * @param valueAlignment The alignment to set for the column values (left, right, or center).
+     * @return The current column object
+     */
+    TableLayout::Column & setValuesAlignment( Alignment valueAlignment );
+
+    /**
+     * @return number of times we will divide the current cell
+     */
+    size_t getNumberCellMerge();
+
+    /**
+     * @brief Increment number of times we will divide the current cell
+     * @param value number of division to add
+     */
+    void incrementMergeHeaderCount( size_t value );
+
+    /**
+     * @brief Decremente number of times we will divide the current cell
+     */
+    void decrementMergeHeaderCount();
+
+    /**
+     * @brief Checks if the column has any child columns.
+     * @return bool True if the column has child columns, otherwise false.
+     */
+    bool hasChild() const;
+
+    /**
+     * @brief Checks if the column has a parent column.
+     * @return bool True if the column has a parent, otherwise false.
+     */
+    bool hasParent() const;
+
+private:
+    /// Pointer to the parent cell (if any).
+    Column * m_parent;
+    /// Pointer to the next cell (if any).
+    Column * m_next;
+    /// The width of the cell (e.g., for cell containing subColumns).
+    size_t m_headerMergeCount  = 0;
   };
 
+  /**
+   * @brief Iterator to loop over all columns, starting by the deepest sub columns,
+   * then to their parents, then to their siblings.
+   */
+  class LeafIterator
```

Comment:
Rename to something more specific to the usage. `LeafIterator` generically tends to denote iterating over all leaf nodes in an arbitrary graph type.

Reply:
I would suggest `Column::LowermostIterator`